src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -890,8 +892,11 @@ JNIEXPORT jobjectArray JNICALL Java_jssc_SerialNativeInterface_waitEvents
         forEnd: {
             returnValues[0] = events[i];
             jintArray singleResultArray = env->NewIntArray(2);
+            if( singleResultArray == NULL ) return NULL;
             env->SetIntArrayRegion(singleResultArray, 0, 2, returnValues);
+            if( env->ExceptionCheck() ) return NULL;
             env->SetObjectArrayElement(returnArray, i, singleResultArray);
```

Comment:
This is probably extremely rare, but in the event that Java DOES throw an exception, does this suppress it?  If so, how useful is it?  Should we raise the exception back to `SerialNativeInterface` or just suppress it?

Reply:
Java exceptions work a bit different when inside a JNI stackframe. They cannot be "thrown" the way as they do in java. This is why I call "*ExceptionCheck()*". It tells us if *SetIntArrayRegion()* has "*thrown*" an exception. If so, we should return so the JVM then has a chance to throw the pending exception into the calling java code.

So in fact this code already is how to "*re-throw*" the exception into the calling java code.

Just for reference in case someone comes along here and would like to read more about the topic, I think [Exception Handling In JNI](https://www.developer.com/open-source/exception-handling-in-jni/) is worth a read.